server/service/transport.go

```diff
@@ -313,7 +314,7 @@ func hostListOptionsFromRequest(r *http.Request) (fleet.HostListOptions, error)
 	case "":
 		// No error when unset
 	default:
-		return hopt, ctxerr.Errorf(r.Context(), "invalid mdm enrollment status %s", enrollmentStatus)
+		return hopt, ctxerr.Wrap(r.Context(), badRequest(fmt.Sprintf("invalid mdm enrollment status %s", enrollmentStatus)))
```

Comment:
Feel free to disregard if you can't address that now, but I think all errors returned by this should be wrapped similarly in `badRequest`.

Reply:
Agree. We should tackle many of those in #4406. (We timed-box this issue as a 1-pointer.)